rrdd/rrdd_server.ml

```diff
@@ -416,29 +417,51 @@ module Plugin = struct
 
 		(* Helper function to find plugin info from a uid. *)
 		let find ~(uid: P.uid) : plugin =
-			Mutex.execute registered_m
-				(fun () -> Hashtbl.find registered uid)
+			try
+				Mutex.execute registered_m (fun () -> Hashtbl.find registered uid)
+			with
+				Not_found ->
+					error "failed to find plugin %s" (P.string_of_uid uid);
+					raise Rrd_protocol.Read_error
```

Comment:
I meant: this is a "find plugin" function. The error here is that is cannot find a plugin in the register. This error can be translated (reraised) into something appropriate for the caller, but "read error" does not make a lot of sense for "find plugin".